doc/guides/device_mgmt/smp_protocol.rst

```diff
@@ -0,0 +1,264 @@
+    |                   |                                                   |
+    +-------------------+---------------------------------------------------+
+    | ``OP``            | :ref:`mcumgr_smp_protocol_op_code`                |
+    |                   |                                                   |
+    +-------------------+---------------------------------------------------+
+    | ``Flags``         | Reserved for flags; there are no flags defined    |
+    |                   | yet, the field should be set to 0                 |
+    |                   |                                                   |
+    +-------------------+---------------------------------------------------+
+    | ``Data Length``   | Length of the ``Data`` field                      |
+    |                   |                                                   |
+    +-------------------+---------------------------------------------------+
+    | ``Group ID``      | This is a command group the frame is intended     |
+    |                   | for.                                              |
+    |                   |                                                   |
+    +-------------------+---------------------------------------------------+
+    | ``Seqence Num``   | This is a frame sequence number.                  |
+    |                   | The number is increased by one with each request  |
+    |                   | frame.                                            |
+    |                   | The Sequence Num of a response should match       |
+    |                   | the one in the request.                           |
+    |                   |                                                   |
+    +-------------------+---------------------------------------------------+
+    | ``Command ID``    | This is a command, within ``Group``.              |
+    |                   |                                                   |
+    +-------------------+---------------------------------------------------+
+    | ``Data``          | This is data payload of the ``Data Length``       |
+    |                   | size. It is optional as ``Data Length`` may be    |
+    |                   | set to zero, which means that no data follows     |
+    |                   | the header.                                       |
+    |                   |                                                   |
+    +-------------------+---------------------------------------------------+
+
+.. note::
+    Contents of a ``Data`` depends on a value of an ``OP``, a ``Group ID``,
+    and a ``Command ID``.
+
+.. _mcumgr_smp_protocol_op_code:
+
+Operation code
+==============
+
+The operation code determines whether an information is written to a device or
+requested from it and whether a packet contains request to a SMP server or
+response from it.
+
+Following operation codes are defined.
```

Comment:
The following operation codes are defined:

Reply:
Done